invenio_app_ils/circulation/api.py

```diff
@@ -204,8 +223,18 @@ def checkout_loan(
         transaction_location_pid=transaction_location_pid,
         transaction_user_pid=transaction_user_pid,
     )
-    pid = ils_circulation_loan_pid_minter(record_uuid, data=new_loan)
-    loan = Loan.create(data=new_loan, id_=record_uuid)
+
+    # check if there is an existing request
+    loan = patron_has_request_on_document(
+        patron_pid, kwargs.get("document_pid")
+    )
+    if loan:
+        loan = loan_cls.get_record_by_pid(loan.pid)
+        pid = IlsCirculationLoanIdProvider.get(loan["pid"]).pid
```

Comment:
isn't it enough to simply do:
```suggestion
        pid = loan.pid
```
?

Reply:
we don't have pid attr on loan, I double checked
<img width="431" alt="Screenshot 2021-05-07 at 11 05 49" src="https://user-images.githubusercontent.com/38131488/117426547-4c606000-af24-11eb-980a-780c119c4209.png">